src/libraries/System.Text.Json/src/System/Text/Json/Schema/JsonSchema.cs

```diff
@@ -0,0 +1,297 @@
+
+                foreach (JsonSchema schema in AnyOf)
+                {
+                    JsonNode schemaNode = schema.ToJsonNode(options);
+                    bool? isTrueOrFalseSchema = ResolveTrueOrFalseSchema(schemaNode);
+
+                    if (isTrueOrFalseSchema is true)
+                    {
+                        // Skip "anyOf" keywords containing true subschemas
+                        anyOfArray = null;
+                        break;
+                    }
+                    else if (isTrueOrFalseSchema is false)
+                    {
+                        // Skip false subschemas
+                        continue;
+                    }
+
+                    anyOfArray.Add(schemaNode);
+                }
+
+                if (anyOfArray != null)
+                {
+                    // Skip "anyOf" keywords containing true schemas
+                    objSchema.Add(AnyOfPropertyName, anyOfArray);
+                }
+            }
+
+            if (HasDefaultValue)
+            {
+                objSchema.Add(DefaultPropertyName, DefaultValue);
+            }
+
+            if (MinLength is int minLength)
+            {
+                objSchema.Add(MinLengthPropertyName, (JsonNode)minLength);
+            }
+
+            if (MaxLength is int maxLength)
+            {
+                objSchema.Add(MaxLengthPropertyName, (JsonNode)maxLength);
+            }
+
+            if (ExporterContext is { } context)
+            {
+                Debug.Assert(options.OnSchemaNodeGenerated != null, "context should only be populated if a callback is present.");
+                // Apply any user-defined transformations to the schema.
+                options.OnSchemaNodeGenerated(context, objSchema);
+            }
+
+            if (ResolveTrueOrFalseSchema(objSchema) is bool trueOrFalse)
+            {
+                // Reduce '{}' or '{ "not": true }' schemas to 'true' and 'false' respectively.
+                return (JsonNode)trueOrFalse;
+            }
+
+            return objSchema;
+
+            static bool? ResolveTrueOrFalseSchema(JsonNode? schema)
+            {
+                switch (schema)
+                {
+                    case JsonValue:
+                        return schema.GetValueKind() switch
+                        {
+                            JsonValueKind.False => false,
+                            JsonValueKind.True => true,
+                            _ => null,
+                        };
+
+                    case JsonObject { Count: 0 }:
+                        return true;
+
+                    case JsonObject { Count: 1 } jsonObject when jsonObject.TryGetPropertyValue(NotPropertyName, out JsonNode? notValue):
+                        return !ResolveTrueOrFalseSchema(notValue);
+
+                    default:
+                        return null;
+                }
+            }
+        }
+
+        private static readonly JsonSchemaType[] s_schemaValues =
```

Comment:
Nit: looks like this could be:
```suggestion
        private static ReadOnlySpan<JsonSchemaType> s_schemaValues =>
```